src/traces/scattercarpet/plot.js

```diff
@@ -0,0 +1,42 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var scatterPlot = require('../scatter/plot');
+var Axes = require('../../plots/cartesian/axes');
+
+module.exports = function plot(gd, plotinfoproxy, data) {
+
+    var carpet = data[0][0].carpet;
+
+    // mimic cartesian plotinfo
+    var plotinfo = {
+        xaxis: Axes.getFromId(gd, carpet.xaxis || 'x'),
+        yaxis: Axes.getFromId(gd, carpet.yaxis || 'y'),
+        plot: plotinfoproxy.plot
+    };
+
+    /* var calcdata = new Array(data.length),
```

Comment:
Haha indeed. There's going to be a lot of review necessary. There's a lot of code here!